code/modules/wiremod/components/ntnet/ntnet_reciever.dm

```diff
@@ -0,0 +1,45 @@
+/**
+ * # NTNet Reciever Component
+ *
+ * Recieves data through NTNet.
+ */
+
+/obj/item/circuit_component/ntnet_receive
+	display_name = "NTNet Receiver"
+	display_desc = "Recieves data packages through NTNet."
+
+	circuit_flags = CIRCUIT_FLAG_OUTPUT_SIGNAL //trigger_output
+
+	network_id = __NETWORK_CIRCUITS
+
+	var/datum/port/input/push_hid
+	var/datum/port/output/hid
+	var/datum/port/output/data_package
+	var/datum/port/output/secondary_package
+
+/obj/item/circuit_component/ntnet_receive/Initialize()
+	. = ..()
+	push_hid = add_input_port("Get HID", PORT_TYPE_SIGNAL)
+	hid = add_output_port("HID", PORT_TYPE_STRING)
+	data_package = add_output_port("Data Package", PORT_TYPE_ANY)
```

Comment:
You probably want this to be a dropdown like the combiner component is, only because "any" isn't actually that *useful*, as far as I know you still can't manipulate it in a lot of ways unless it's primitive (like to string).